### PR TITLE
pulp-test: Allow passthrough flags to test program

### LINKED ISSUE #239: pulp test does not pass arguments to test programs correctly
Currently you can't pass psa flags to pulp test, ie:

`pulp test --stash --censor-lib --strict`

### PR DESCRIPTION
**Pull request by mcoffin**
Fixes #239 

Duplicate of #250 w/ new code. GitHub didn't like that I force-pushed to the old branch and wouldn't let me re-open.

### CODE REVIEW

File: test-js/integration.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import which from "which";
 
 const hello = "Hello sailor!";
 const test = "You should add some tests.";
+const testArguments = "--something-node-wouldnt-like"
 const docLine1 = "## Module Main";
```
Comment (hdgarrood) [inline test-js/integration.js]: Is this used?

File: test-js/integration.js [hunk 2]
```diff
@@ -236,6 +237,17 @@ describe("integration tests", function() {
     assert.equal(out.trim(), test);
   }));
 
+  it("pulp test --test-path test2 -- --something-node-wouldnt-like", run(function*(sh, pulp, assert, temp) {
+    const newTest = path.join(temp, "test2");
+    yield pulp("init");
+    yield fs.mkdir(newTest);
+    yield fs.copy(path.resolve(__dirname, "resources/Main.purs"), path.join(newTest, "Main.purs"));
+    yield fs.copy(path.resolve(__dirname, "resources/Main.js"), path.join(newTest, "Main.js"));
+    yield sh("bower i purescript-arrays --save");
```
Comment (hdgarrood) [inline test-js/integration.js]: Can you think of a good way of avoiding the need to invoke `bower` here? I'd like to make it as easy as possible to switch to `psc-package` once we are ready to.

**Review by hdgarrood — COMMENTED** [review]
Looks great, just a couple of minor comments :)

**Comment by mcoffin** [comment]
@hdgarrood that should do it (f450b54).

**Comment by hdgarrood** [comment]
Looks great, thanks!

**Pull request merged**